spec/src/main/asciidoc/repository.asciidoc

```diff
@@ -353,26 +355,32 @@ The following examples work through scenarios where static and dynamic sort crit
 [source,java]
 ----
 // Sorts first by type. When type is the same, applies the Order's sort criteria
-Page<User> findByNameStartsWithOrderByType(String namePrefix,
-                                           PageRequest pagination,
-                                           Order<User> sorts);
+@Query("where left(name,length(?1)) = ?1")
+@OrderBy(_User.TYPE)
+Page<User> findByNamePrefix(String namePrefix,
+                            PageRequest pagination,
+                            Order<User> sorts);
```

Comment:
Prior to your changes, these examples covered 5 scenarios:

OrderBy keyword + Order param
OrderBy keyword + Sort... param
OrderBy anno + Order param
OrderBy anno + Sort... param
Query without ORDER BY + OrderBy anno + Order param

After changes, it now only covers 2 scenarios with several duplicated:

Query without ORDER BY + OrderBy anno + Order param
Query without ORDER BY + OrderBy anno + Sort... param
Query without ORDER BY + OrderBy anno + Order param
Query without ORDER BY + OrderBy anno + Sort... param
Query without ORDER BY + OrderBy anno + Order param

Adding suggestions to restore to previous:

```suggestion
Page<User> findByNameStartsWithOrderByType(String namePrefix,
                                           PageRequest pagination,
                                           Order<User> sorts);
```